testutil/integration/evmos/utils/gov.go

```diff
@@ -125,3 +163,32 @@ func getProposalIDFromEvents(events []abcitypes.Event) (uint64, error) {
 
 	return proposalID, nil
 }
+
+func submitProposal(tf factory.TxFactory, network network.Network, proposerPriv cryptotypes.PrivKey, txArgs commonfactory.CosmosTxArgs) (uint64, error) {
+	res, err := tf.ExecuteCosmosTx(proposerPriv, txArgs)
+	if err != nil {
+		return 0, err
+	}
+
+	proposalID, err := getProposalIDFromEvents(res.Events)
+	if err != nil {
+		return 0, errorsmod.Wrap(err, "failed to get proposal ID from events")
+	}
+
+	err = network.NextBlock()
+	if err != nil {
+		return 0, errorsmod.Wrap(err, "failed to commit block after proposal")
+	}
+
+	gq := network.GetGovClient()
+	proposalRes, err := gq.Proposal(network.GetContext(), &govv1.QueryProposalRequest{ProposalId: proposalID})
+	if err != nil {
+		return 0, errorsmod.Wrap(err, "failed to query proposal")
+	}
+
+	if proposalRes.Proposal.Status != govv1.StatusVotingPeriod {
+		return 0, fmt.Errorf("expected proposal to be in voting period; got: %s", proposalRes.Proposal.Status.String())
+	}
```

Comment:
ditto - could be the same function on both but receiving a status params to check against and returns a value to see if its equal to the expected status